### ISSUE TITLE
Enforce blank line before and after comments

### ISSUE THREAD

**Issue by wincent** [liferay/liferay-npm-tools#409]
# ~SCSS~

~[Requested by Brian here](https://github.com/brianchandotcom/liferay-portal/pull/85868#issuecomment-595881597).~

~[Sample format](https://github.com/brianchandotcom/liferay-portal/commit/679ae67ab8b6aa3650f924e8f452a3cb146da89e).~

(SCSS part of this split off into https://github.com/liferay/liferay-npm-tools/issues/475)

# JS

[Requested by Brian here](https://github.com/brianchandotcom/liferay-portal/pull/87232#issuecomment-608623839).

[Sample format](https://github.com/brianchandotcom/liferay-portal/commit/9611cccb243732aac4a839058c25da867c070cbe).

Related: https://eslint.org/docs/rules/lines-around-comment

**Comment by bryceosterhaus** [comment]
So does this mean we need to add more custom rules to prettier?  my fears are becoming reality lol

**Comment by wincent** [comment]
No, I believe this would be adding custom rules to style-lint. So your fears are not yet realized! 😂

**Comment by wincent** [comment]
Sadly, seems that Prettier does disagree about blank lines above comments at the top of a new scope ([example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAjCATAngAmAHSkJxwHpScB3AQxkOJwDoAzCCPBkyAGwgCckOAMTNRAbgYBfQpJAAaEBAAOMAJbQAzslDU+fCJQAKuhFpTUAbhFUZ5aPtTABrODADKSx6qgBzZDD4AVzgFAAsYAFtuAHVQ1XgNTzA4N1N41Qt4rGRwDS0Fbw04PhhDBx8I6mRmam4ihQArDQAPACEHZ1c3agi4ABlvOGra+pAm5rdvH244AEVAiHhhupCQTz4ivhyNMDy7JT5vGGibGFDkAA4ABgUDiCLohyUcg7hNiyGFAEcF+DLlMwgagaAC0UDgcAwkLsfDgP1UsLK1AqVSQNRWCiKEVU-iCqw0Uxm80WQzRI1WMGoqBOGDOyAATAoAtRVNwpgBhCARSo5KDQT4gQJFAAqVLM6NGFmCAEkoFDYG4wIcVABBOVuGBYGbLIqSSRAA)), so I suggest we consider this one a very low priority (because the cost is ⬆️ relative to the benefit ⬇️).

**Comment by wincent** [comment]
Updated the description for this to reflect that we got [another request](https://github.com/brianchandotcom/liferay-portal/pull/87232#issuecomment-608623839) (this time [in JS](https://github.com/brianchandotcom/liferay-portal/commit/9611cccb243732aac4a839058c25da867c070cbe)).

Unlike the first example, which would have conflicted with Prettier, [this one doesn't](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdGBLGAKA5AJwQBM58ACAQzKPwgAciIB3KMgWwQFcymsALMjD5wykfITAwyEfCXy4ANGWwBKMgF4AfGWCoo6dAHpDZAHTm96AL4qA3Jf0xjZADIAFAMoBaAIx+ALAAcgQ7oWHiEUHKU1LQMzKwcUNy8QoLCojISUjJyispqWjqWMEYm5qYlNrYgCiD0mNAAzsigFOLMbu0ILSgUADZMFACeLXUARvgUYADWcDAedNMYUADmyDD4nHB1fDBs-QDqfFhwTUtgcB49WBgAbljDyOBNYyArTaQwblOrbBTIABmA0+dQAVk0AB4AISms3mHgoHBcKzgQJBOxAEMhHhWq36cAAipwIPB0f1QSAlvhPvhnuMKOM4P1alT8CsYIcMEQhMhAgAGOp0WifQ5TOjPYVnUh3NF1ACOJPgP3ovRAFCaXigcDgJCIrMIiowhB+FD+AKQwIpmM+bAwGy2NrxBOJpLRloxdRgjK5PL4yAATF6phh+niAMIQNj-Z5nACsrM4nwAKozelbKXdtgBJKIIBZgdl0GAAQSiHhgwwJ5M+VisQA):

```js
	it('render a dropdown menu with the correct order', () => {
		// ...
	});

	// LPS-111488

	it('render a dropdown menu with the correct order', () => {
		// ...
	});
```

In other words, Prettier is going to respect blank lines before and after comments *except* comments that are the first thing in a new scope. This is true for both SCSS and JS. That means we can get 90%(?) of the way there without any hackery, so we may as well do that, and defer the decision on whether its worth going the remaining 10% — it might not even come up that often.

**Comment by wincent** [comment]
Another request here: https://github.com/brianchandotcom/liferay-portal/pull/87372#issuecomment-610716460

And seeing as this request now covers both JS and SCSS, I created a ticket in eslint-config-liferay for the JS part that will be handled by ESLint: https://github.com/liferay/eslint-config-liferay/issues/169

**Comment by wincent** [comment]
JS part is prepared over here: https://github.com/liferay/eslint-config-liferay/pull/170

**Comment by wincent** [comment]
So JS side of this was done a while back, and I created (duplicate) https://github.com/liferay/liferay-npm-tools/issues/475 for the SCSS side because I forgot about this ticket.

May as well close this one and keep the other one for SCSS. I edited the description to reflect that.
